contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,952 @@
+        // governance transaction batch. In that case the WETH that is on the strategy contract
+        // also needs to be accounted for.
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        // just paranoia check, in case there is OETHb in the strategy that for some reason hasn't
+        // gotten burned yet.
+        uint256 oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        return underlyingAssets + wethBalance + oethbBalance;
+    }
+
+    /**
+     * @dev Returns the balance of both tokens in a given position (excluding fees)
+     * @return amountWeth Amount of WETH in position
+     * @return amountOethb Amount of OETHb in position
+     */
+    function getPositionPrincipal()
+        public
+        view
+        returns (uint256 amountWeth, uint256 amountOethb)
+    {
+        if (tokenId == 0) {
+            return (0, 0);
+        }
+
+        uint160 sqrtRatioX96 = getPoolX96Price();
+        (amountWeth, amountOethb) = helper.principal(
+            positionManager,
+            tokenId,
+            sqrtRatioX96
+        );
+    }
+
+    function getPoolX96Price() public view returns (uint160 sqrtRatioX96) {
+        (sqrtRatioX96, , , , , ) = clPool.slot0();
+    }
+
+    /**
+     * @dev get total amount of underlying pool tokens
+     */
+    function _getTotalPoolTokens() internal view returns (uint256 totalTokens) {
+        totalTokens =
+            IERC20(WETH).balanceOf(address(clPool)) +
+            IERC20(OETHb).balanceOf(address(clPool));
+    }
+
+    function _getLiquidity() internal view returns (uint128 liquidity) {
+        if (tokenId == 0) {
+            revert("No LP position");
+        }
+
+        (, , , , , , , liquidity, , , , ) = positionManager.positions(tokenId);
+    }
+
+    /***************************************
+            Hidden functions
+    ****************************************/
+    /// @inheritdoc InitializableAbstractStrategy
+    function setPTokenAddress(address, address) external override {
+        // The pool tokens can never change.
+        revert("Unsupported method");
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function removePToken(uint256) external override {
+        // The pool tokens can never change.
+        revert("Unsupported method");
+    }
+
+    /***************************************
+            ERC721 management
+    ****************************************/
+
+    /// @notice Callback function for whenever a NFT is transferred to this contract
+    //  solhint-disable-next-line max-line-length
+    /// Ref: https://docs.openzeppelin.com/contracts/3.x/api/token/erc721#IERC721Receiver-onERC721Received-address-address-uint256-bytes-
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes calldata
+    ) external returns (bytes4) {
+        return this.onERC721Received.selector;
```

Comment:
⚪ This seems to be blindly accepting all NFTs. So, someone can mint their liquidity position and transfer that NFT to the contract. However we use `tokenId` everywhere, so it shouldn't cause any problem.

Reply:
good point. As you point out I also see no problem with collecting various NFTs